### PR TITLE
Configure spring

### PR DESCRIPTION
**Pull request by yutsevich**

### CODE REVIEW

**Review by andriylagun — APPROVED** [review]
well done

File: src/main/java/com/dev/cinema/dao/AbstractDao.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import com.dev.cinema.dao.impl.CinemaHallDaoImpl;
+import com.dev.cinema.exceptions.DataProcessingException;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class AbstractDao<T> {
+    private static final Logger logger = Logger.getLogger(CinemaHallDaoImpl.class);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/AbstractDao.java]: CinemaHallDaoImpl.class -?

File: src/main/java/com/dev/cinema/dao/AbstractDao.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    protected final SessionFactory sessionFactory;
+
+    @Autowired
+    public AbstractDao(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    public T addAbstract(T entity) {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/AbstractDao.java]:
```suggestion
    public T create/add(T entity) {
```

File: src/main/java/com/dev/cinema/dao/AbstractDao.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+import com.dev.cinema.dao.impl.CinemaHallDaoImpl;
+import com.dev.cinema.exceptions.DataProcessingException;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class AbstractDao<T> {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/AbstractDao.java]: what do you think about abstract? Should we make this class abstract?

File: src/main/java/com/dev/cinema/dao/OrderDao.java [hunk 4]
```diff
@@ -1,12 +1,11 @@
...
-import com.dev.cinema.model.Ticket;
 import com.dev.cinema.model.User;
 import java.util.List;
 
 public interface OrderDao {
-    Order add(List<Ticket> tickets, User user);
-
     List<Order> getByUser(User user);
+
+    void add(Order order);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/OrderDao.java]: why do we return void? and in other interfaces too?

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**